recipes/premake/5.x/conanfile.py

```diff
@@ -11,47 +12,109 @@ class PremakeConan(ConanFile):
     homepage = "https://premake.github.io"
     license = "BSD-3-Clause"
     settings = "os", "arch", "compiler", "build_type"
-    _source_subfolder = "source_subfolder"
+    exports_sources = "patches/**"
+    options = {
+        "lto": [True, False],
+    }
+    default_options = {
+        "lto": False,
+    }
 
-    def configure(self):
-        if self.settings.os == "Windows" and self.settings.compiler == "gcc":
-            raise ConanInvalidConfiguration("Building with MinGW isn't supported currently by the recipe")
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+    def config_options(self):
+        if self.settings.os != "Windows" or self.settings.compiler == "Visual Studio":
+            del self.options.lto
+
+    @property
+    def _msvc_version(self):
+        return {
+            "12": "2013",
+            "14": "2015",
+            "15": "2017",
+            "16": "2019",
+        }.get(str(self.settings.compiler.version), "2017")
+
+    @property
+    def _msvc_build_dirname(self):
```

Comment:
IIRC correctly this won't work for all versions as different Premake versions offer different VS build solutions

Reply:
Checking again. alpha14 and alpha15 offer the same options. alpha13 and alpha12 did not offer vs2019, but we don't have it in CCI (we had it in [Bincrafters](https://github.com/bincrafters/conan-premake_installer)).

Anyhow, this is probably something which needs to be tested when different versions get added